src/main/java/cloud/fogbow/ras/core/OrderController.java

```diff
@@ -157,6 +157,37 @@ public void deleteOrder(Order order) throws FogbowException {
         }
     }
 
+    public void takeSnapshot(Order order, String name, SystemUser systemUser) throws FogbowException {
+        synchronized (order){
+            OrderState orderState = order.getOrderState();
+
+            if (orderState.equals(OrderState.CHECKING_DELETION) ||
```

Comment:
The code below is for delete method. Here you should check whether the order is in the correct state (check in the documentation of the various clouds whether the VM needs to be stopped for a snapshot to be taken). If it is not in the correct state, then an UnacceptableOperationException must be thrown with an appropriate message.